### ISSUE TITLE
Improve Code Modularity

### ISSUE THREAD

**Issue by CracktheDom** [sockheadrps/PyProfileDataGen#3]
The functions are well-separated, you could consider breaking down the functions into smaller ones (e.g. one for reading the README file, another for writing to the README file, another finding the insertion point) for clarity and easier maintenance.

**Comment by sockheadrps** [comment]
Yeah this could definitely benefit from a refactor. The readme script in particular is a little all over the place.
